### DISCUSSION TITLE
quark 离线下载功能的实现讨论

### DISCUSSION THREAD

**Discussion by Vizards** [AlistGo/alist#2585] [Ideas]
首先定义一下问题：这里的离线下载是指其**桌面客户端**上提供的**离线下载到网盘内**的功能，不是 aria2 的离线下载。如下图。
<img width="463" alt="image" src="https://user-images.githubusercontent.com/13443193/205568393-d577c7a3-02a0-495b-9eef-3aedd7a5b11a.png">

目前已经通过 decompiling 的方式知晓了客户端的实现方式并通过简单 demo 验证，所以咨询下是否有必要在 Alist 内增加此功能？
现在能想到的几个问题：

1. 风险问题，客户端是通过加密 cookie 和请求体来阻止用户通过接口直接使用此功能的；
2. 与 Alist 本身功能是否契合，本质上它和网盘管理功能的关系不大；
3. 夸克的 cookie 失效和重新获取逻辑目前不明确，可能对此功能的稳定性造成影响。

**Comment by NovaShen0724** [comment]
aria2都加上了，(2)就不用担心了，可以测试夸克的cookie有效期以及失效的触发事件，如果能上线，一定会超越aria2的使用
